arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,368 @@
+
+Let's convert our dumb contract into a smart contract.
+
+
+### Configure your project directory
+
+Create a `decentralized-cupcakes` directory for your project and install [hardhat](https://hardhat.org/hardhat-runner/docs/getting-started#overview)[^6] using VS Code's integrated terminal:
+
+```
+mkdir decentralized-cupcakes
+cd decentralized-cupcakes
+yarn init -y
+yarn add hardhat @nomicfoundation/hardhat-toolbox
+```
+
+This installs two packages: `hardhat` lets us test and deploy our smart contracts, and `hardhat-toolbox` is a bundle of popular Hardhat plugins that we'll use later.
+
+Next, run `npx hardhat` to configure Hardhat. Select `Create a JavaScript project` when prompted. Make sure you specify your `decentralized-cupcakes` directory as the project root when asked.
+
+At this point, you should see the following items (among others) in your `decentralized-cupcakes` project directory:
+
+| Item                | Description                                                                                               |
+| ------------------- | --------------------------------------------------------------------------------------------------------- |
+| `contracts/`        | Contains your smart contracts. You should see the `Lock.sol` contract here.                               |
+| `scripts/`          | Contains scripts that you can use to interact with your smart contracts. You should see `deploy.js` here. |
+| `hardhat.config.js` | Contains the configuration settings for Hardhat.                                                          |
+
+
+Replace the contents of `hardhat.config.js` with the following[^6]:
+
+```javascript title="hardhat.config.js"
+require("@nomicfoundation/hardhat-toolbox");
+
+/** @type import('hardhat/config').HardhatUserConfig */
+module.exports = {
+  solidity: "0.8.18",
+  networks: {
+    hardhat: {
+      chainId: 1337
+    },
+    arbitrumGoerli: {
+      url: "https://goerli-rollup.arbitrum.io/rpc",
+      chainId: 421613
+      //accounts: [process.env.PRIVATE_KEY] TODOs
+    },
+    arbitrumOne: {
+      url: "https://arb1.arbitrum.io/rpc",
+      //accounts: [process.env.PRIVATE_KEY]
+    },
+    arbitrumNova: {
+      url: "https://arbnova.com/rpc",
+      //accounts: [process.env.PRIVATE_KEY]
+    }
+  }
+};
+```
+
+Run `npx hardhat compile` to compile the default `contracts`. You may be prompted to install additional dependencies - follow those instructions until this command runs successfully[^6]. You should see `Compiled 1 Solidity file successfully` in the terminal output. You should also see a new `decentralized-cupcakes/artifacts/` directory. This directory contains the compiled smart contract[^7].
+
+Open `scripts/deploy.js` and replace its contents with the following:
```

Comment:
Is it weird if I find it confusing to edit deploy.js before VendingMachine.sol ? I'm thinking about following a logic similar to "set config => write contract => deploy it". Disregard if it looks clear enough as it is.

Reply:
It's not weird it's a wonderful nudge, updating ^^

ty 🙏

Reply:
Ended up on this order of operations:

1. Review JS
2. Configure project directory (all hardhat concerns here - installation and config)
3. Compile defaults
4. Compile VendingMachine
5. Deploy locally
  1. Run hardhat
  2. Configure metamask (all metamask concerns here)
  3. Deploy & test
6. Update config, deploy to testnet
7. Update config, deploy to mainnet